### PR TITLE
fix(kafka): add clusterName because of api-changes

### PR DESCRIPTION
**Pull request by haarchri**
Signed-off-by: Christopher Paul Haar <christopherpaul.haar@dkb.de>

### Description of your changes
fix(kafka): add clusterName because of api-changes

Fixes #1612
replace: #1620

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
```
kubectl get managed

NAME                                      READY   SYNCED   EXTERNAL-NAME
cluster.kafka.aws.crossplane.io/example   True   True     arn:aws:kafka:us-east-1:12345678910:cluster/example/cef294f9-8e22-4342-9bfd-9635fcaeaebb-9

NAME                                                          READY   SYNCED   EXTERNAL-NAME
configuration.kafka.aws.crossplane.io/example-configuration   True    True     arn:aws:kafka:us-east-1:12345678910:configuration/example-configuration/73a88bf4-8aee-485e-b004-a788735643d4-9
```

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: package/crds/kafka.aws.crossplane.io_clusters.yaml [hunk 1]
```diff
@@ -479,6 +482,7 @@ spec:
                     description: Create tags when creating the cluster.
                     type: object
                 required:
+                - clusterName
```
Comment (chlunde) [inline package/crds/kafka.aws.crossplane.io_clusters.yaml:485]: would defaulting this parameter to the object name provide backwards compatibility? Would it make sense?

Comment (haarchri) [reply package/crds/kafka.aws.crossplane.io_clusters.yaml:485]: problem is for people using compositions - the cr.name is complete different

**Review by chlunde — APPROVED** [review]

**Comment by abrarcci** [comment]
FYI - have tested out the new changes @haarchri and the msk cluster is now being provisioned, thank you!

One point to note, doing a describe on the cluster gives the following:

```
  Warning  CannotObserveExternalResource  3m22s (x60 over 58m)  managed/cluster.kafka.aws.crossplane.io  failed to describe Cluster: InvalidParameter: 1 validation error(s) found.
- missing required field, DescribeClusterInput.ClusterArn.

```

And the status of the cluster never changes from this:

```
Status:
  At Provider:
  Conditions:
    Last Transition Time:  2023-01-17T12:50:14Z
    Reason:                Creating
    Status:                False
    Type:                  Ready
    Last Transition Time:  2023-01-17T12:50:14Z
    Message:               observe failed: failed to describe Cluster: InvalidParameter: 1 validation error(s) found.
- missing required field, DescribeClusterInput.ClusterArn.
```

**Comment by haarchri** [comment]
Yes we found the same we will update shortly

**Comment by haarchri** [comment]
we will publish tomorrow a new fix release  https://github.com/crossplane-contrib/provider-aws/issues/1638

**Pull request merged**